app/models/tagged_animal_assessment.rb

```diff
@@ -45,21 +45,28 @@ class TaggedAnimalAssessment < ApplicationRecord
     NOTES: "Notes",
   }
 
+  # this is used to dynamically define argument setter for these attributes
+  DATE_ATTRIBUTES = %w[
+    measurement_date
+    spawning_date
+  ].freeze
+
   validates :measurement_date, :shl_case_number, :spawning_date, :tag, :length, presence: true
   validates :length, numericality: true
 
   def self.create_from_csv_data(attrs)
     new(attrs)
   end
 
-  def measurement_date=(measurement_date_str)
-    return unless measurement_date_str
-    write_attribute(:measurement_date, DateTime.strptime(measurement_date_str, '%m/%d/%y'))
-  end
-
-  def spawning_date=(spawning_date_str)
-    return unless spawning_date_str
-    write_attribute(:spawning_date, DateTime.strptime(spawning_date_str, '%m/%d/%y'))
+  DATE_ATTRIBUTES.each do |name|
+    define_method "#{name}=" do |argument|
+      return unless argument
+      begin
+        write_attribute(name.to_sym, DateTime.strptime(argument, "%m/%d/%y"))
+      rescue ArgumentError
+        errors.add(name.to_sym, :invalid, message: "Invalid date format: #{argument}")
+      end
+    end
```

Comment:
@edwinthinks I think this should address your concern regarding Date parsing errors.

I'm not sure if there's a better way to do this? Or to dry it up some more? Because I have codes repeating for tagged and untagged models.

I tried extracting the metaprogramming part out into a module like so:

```
module DateAssignment
  extend ActiveSupport::Concern

  DATE_ATTRIBUTES.each do |name|
    ...
   end
end
```

but can't seem to get the constant's scope figured out.